src/opts.c

```diff
@@ -695,17 +695,18 @@ natsOptions_SkipServerVerification(natsOptions *opts, bool skip)
     if (s == NATS_OK)
     {
         opts->sslCtx->skipVerify = skip;
-        if (skip)
-        {
-            opts->sslCtx->callback = NULL;
-        }
+#ifdef NATS_WITH_EXPERIMENTAL
+        opts->sslCtx->callback = NULL;
```

Comment:
Why removing the check for `skip`?

Reply:
Yes, but silly example, but say user sets the callback first (which sets skip to false internally and sets the callback), then this API is called with skip=false, it would be a no-op, but in this case it will now set skip to false AND delete the callback...